File: 백준/Silver/10828. 스택/스택.py
import sys
input = sys.stdin.readline



class fixedStack:
    def __init__(self):
        self.stk=[]
        self.length=0

    def push(self,value):
        self.stk.append(value)
        self.length+=1
    
    def pop(self):
        if self.length==0:
            print(-1)
        else:
            print(self.stk.pop())
            self.length-=1

    def size(self):
        print(self.length)

    def empty(self):
        if self.length>0:
            print(0)
        else:
            print(1)

    def top(self):
        if self.length<=0:
            print(-1)
        else:
            print(self.stk[-1])
            
            
n=int(input())
stack = fixedStack()

for _ in range(n):
    command=list(map(str,input().split()))

    if command[0] == 'push':
        stack.push(int(command[1]))
    elif command[0] == 'pop':
        stack.pop()
    elif command[0]=='size':
        stack.size()
    elif command[0]=='empty':
        stack.empty()
    elif command[0]=='top':
        stack.top()

File: 백준/Silver/10828. 스택/스택.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Map;
import java.util.StringTokenizer;

public class Main {
    private static StringBuilder sb = new StringBuilder();
    private static class Stack {

        int[] stk = new int[100_000];
        int size = 0;

        void push(int n){
            this.stk[this.size++] = n;
        }

        void pop() {
            if(this.size==0)
                sb.append(-1);
            else
                sb.append(this.stk[--this.size]);
            sb.append("\n");
        }

        void size(){
            sb.append(this.size);
            sb.append("\n");
        }

        void empty() {
            if (this.size==0)
                sb.append(1);
            else
                sb.append(0);
            sb.append("\n");
        }

        void top() {
            if(this.size==0)
                sb.append(-1);
            else
                sb.append(this.stk[size-1]);
            sb.append("\n");
        }
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        Stack stk = new Stack();

        int n = Integer.parseInt(st.nextToken());
        String order; int num;
        for (int i = 0; i < n; i++) {
            st = new StringTokenizer(br.readLine());
            order = st.nextToken();
            switch (order){
                case "push":
                    num = Integer.parseInt(st.nextToken());
                    stk.push(num);
                    break;
                case "pop":
                    stk.pop();
                    break;
                case "size":
                    stk.size();
                    break;
                case "empty":
                    stk.empty();
                    break;
                case "top":
                    stk.top();
                    break;
            }
        }
        System.out.println(sb.toString());
    }
}
